test/jdk/tools/jpackage/macosx/SigningAppImageTest.java

```diff
@@ -74,4 +75,22 @@ public static void test() throws Exception {
         SigningBase.verifyCodesign(appImage, true);
         SigningBase.verifySpctl(appImage, "exec");
     }
+
+    @Test
+    public static void testAdditionalLauncher() throws Exception {
+        SigningCheck.checkCertificates();
+
+        JPackageCommand cmd = JPackageCommand.helloAppImage();
+        cmd.addArguments("--mac-sign", "--mac-signing-key-user-name",
+                SigningBase.DEV_NAME, "--mac-signing-keychain",
+                SigningBase.KEYCHAIN);
+        AdditionalLauncher testAL = new AdditionalLauncher("testAL");
+        testAL.applyTo(cmd);
+
+        cmd.executeAndAssertHelloAppImageCreated();
+        Path launcherPath = cmd.appLauncherPath();
+        Path testALPath = launcherPath.getParent().resolve("testAL");
+        SigningBase.verifyCodesign(launcherPath, true);
+        SigningBase.verifyCodesign(testALPath, true);
+    }
```

Comment:
Would it make sense to extend the existing test with additional launcher checks instead of creating another one? Packaging of a full JDK with signing probably takes a while

Reply:
yes - that would be simpler - will do.